### PR TITLE
Enhance Utils.deserialize, fix EE producer test and re-enable it.

### PR DESCRIPTION
**Pull request by manovotn**
Will **not pass** until we run it against WFLY 11 as it contains fix for WFLY-835.

### CODE REVIEW

File: tests-common/src/main/java/org/jboss/weld/test/util/Utils.java [hunk 1]
```diff
@@ -177,12 +183,19 @@ public static boolean isProxy(Object proxy) {
     private static class TCCLObjectInputStream extends ObjectInputStream {
 
         private final ClassLoader classLoader;
+        private ClassLoader lastResortClassLoader = null;
```
Comment (mkouba) [inline tests-common/src/main/java/org/jboss/weld/test/util/Utils.java]: `lastResortClassLoader`? Choose a better name pls :D The original `classLoader` could be renamed to `tccl` or something like that.Also the field should be final.

File: tests-common/src/main/java/org/jboss/weld/test/util/Utils.java [hunk 2]
```diff
@@ -177,12 +183,19 @@ public static boolean isProxy(Object proxy) {
...
 
         public TCCLObjectInputStream(InputStream in) throws IOException {
             super(in);
             this.classLoader = Thread.currentThread().getContextClassLoader();
```
Comment (mkouba) [inline tests-common/src/main/java/org/jboss/weld/test/util/Utils.java]: Replace with `this(in, null)`

File: tests-common/src/main/java/org/jboss/weld/test/util/Utils.java [hunk 3]
```diff
@@ -192,7 +205,17 @@ public TCCLObjectInputStream(InputStream in) throws IOException {
...
-                    return BeanManagerImpl.class.getClassLoader().loadClass(desc.getName());
+                    try {
+                        return BeanManagerImpl.class.getClassLoader().loadClass(desc.getName());
+                    } catch (ClassNotFoundException cnfe) {
+                        if (lastResortClassLoader != null) {
+                            // should all else fail, try the alternatively supplied CL
+                            return lastResortClassLoader.loadClass(desc.getName());
+                        } else {
+                            // rethrow the exception, we cannot handle this
+                            throw new ClassNotFoundException(cnfe.getMessage(), cnfe.getCause());
```
Comment (mkouba) [inline tests-common/src/main/java/org/jboss/weld/test/util/Utils.java]: Why not simply `throw cnfe`?

**Comment by manovotn** [comment]
Adjusted accordingly to comments, I also put `@Ignore` on the test with a comment.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**